### DISCUSSION TITLE
waitForTransactionReceipt timeout exception after confirmed

### DISCUSSION THREAD

**Discussion by boyd-dev** [wevm/viem#1351] [Question] [answered]
```
const [receipt, setReceipt] = useState(null);
const publicClient = usePublicClient();

if (txHash) {
     publicClient.waitForTransactionReceipt({ hash: txHash, timeout: 36_000 }).then(v => setReceipt(v));
}
```
`waitForTransactionReceipt` returns the receipt in a few seconds but after 36 seconds it says:

```
Timed out while waiting for transaction with hash "0x6a2...7a77" to be confirmed.

Version: viem@1.16.5
WaitForTransactionReceiptTimeoutError: Timed out while waiting for transaction with hash "0x6a2...7a77" to be confirmed.
```

I think it should stop to wait after resolving the result.  What could possibly wrong?

**Comment by JakubGrzegowski** [comment]
I think I got a similar issue, the tx is being sent successfully, however the waitForTransactionReceipt function seems to be still be executing afterward, eventually dropping the timeout exception if configured to do so.

**Comment by FlakelfXYZ** [comment]
Any updates?

**Comment by melzsanbox** [comment]
@jxom please answer

**Reply by jxom** [reply, marked as answer]
https://github.com/wevm/viem/issues/new?assignees=&labels=&projects=&template=bug_report.yml
